project_import_export.go

```diff
@@ -173,10 +179,41 @@ type ImportFileOptions struct {
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/project_import_export.html#import-a-file
 func (s *ProjectImportExportService) ImportFile(opt *ImportFileOptions, options ...RequestOptionFunc) (*ImportStatus, *Response, error) {
-	req, err := s.client.NewRequest(http.MethodPost, "projects/import", opt, options)
+	file, err := os.Open(*opt.File)
 	if err != nil {
 		return nil, nil, err
 	}
+	defer file.Close()
+	var b bytes.Buffer
+	writer := multipart.NewWriter(&b)
+	var optMap map[string]interface{}
+	optMarshalled, err := json.Marshal(opt)
+	if err != nil {
+		return nil, nil, err
+	}
+	json.Unmarshal(optMarshalled, &optMap)
+
+	for key, value := range optMap {
+		if strings.ToLower(key) == "file" {
+			part, err := writer.CreateFormFile("file", file.Name())
+			if err != nil {
+				return nil, nil, err
+			}
+			_, err = io.Copy(part, file)
+		} else {
+			_ = writer.WriteField(strings.ToLower(key), fmt.Sprintf("%v", value))
+		}
+	}
```

Comment:
No clue what you are trying to do here, but this doesn't look right to me. Can you please elaborate on this approach?

Reply:
Sure, we're creating a multipart form, creating the file component with CreateFormFile(), copying the file contents to that part, then adding each of the fields from the ImportFileOptions struct as additional field data in the multipart form.  That's the intent, anyway.  I'll admit to being n00bish with golang, so maybe this isn't the proper way to do the struct-to-string-map conversion, but that's the basic idea.

Reply:
Dunno if that answered your question, happy to clarify further if need be.